File: ch8/py/d-authorize.py
from langchain.schema import HumanMessage
from langgraph.graph import StateGraph
from langgraph.checkpoint.memory import MemorySaver


async def main():
    # Create a simple graph
    builder = StateGraph()
    # Add nodes and edges as needed
    graph = builder.compile(checkpointer=MemorySaver())

    input = {
        "messages": [
            HumanMessage(
                "How old was the 30th president of the United States when he died?"
            )
        ]
    }

    config = {"configurable": {"thread_id": "1"}}

    output = graph.astream(input, config, interrupt_before=["tools"])

    async for c in output:
        print(c)  # do something with the output


if __name__ == "__main__":
    import asyncio

    asyncio.run(main())

File: ch8/js/d-authorize.js
import { HumanMessage } from "@langchain/core/messages";
import { MemorySaver } from "@langchain/langgraph";
import { StateGraph } from "@langchain/langgraph";
import { Annotation } from "@langchain/langgraph";
import * as dotenv from "dotenv/config";
dotenv;

const GraphState = Annotation.Root({
  messages: Annotation(),
});

// Assuming graph is already created and configured
const graph = new StateGraph(GraphState).compile({
  checkpointer: new MemorySaver(),
});

const input = {
  messages: [
    new HumanMessage(
      "How old was the 30th president of the United States when he died?"
    ),
  ],
};

const config = { configurable: { thread_id: "1" } };

const output = await graph.stream(input, {
  ...config,
  interruptBefore: ["tools"],
});

console.log("Streaming output:", output);
for await (const chunk of output) {
  console.log(chunk); // do something with the output
}
